test/categorical.jl

```diff
@@ -6,6 +6,7 @@ using Test
 for p in Any[
     [0.5, 0.5],
     [0.5f0, 0.5f0],
+    [1//2, 1//2],
```

Comment:
The modified tests don't test the new generic zero
`pdf(d::DiscreteUnivariateDistribution, x::Real) = isinteger(x) ? pdf(d, round(Int, x)) : zero(partype(d))`

Reply:
I believe they do. The test in https://github.com/JuliaStats/Distributions.jl/blob/c89d557344591a2c046280b96b2597d019457583/test/categorical.jl#L28 will dispatch to https://github.com/JuliaStats/Distributions.jl/blob/c89d557344591a2c046280b96b2597d019457583/src/univariates.jl#L342 which will then dispatch to the modified version of https://github.com/JuliaStats/Distributions.jl/blob/c89d557344591a2c046280b96b2597d019457583/src/univariates.jl#L333. I wrote the test before the fix, so I think it's exercised.